executor/executor.go

```diff
@@ -160,3 +163,31 @@ func GetWorkflowDataVersion(context context.Context, workflowID string, sdb *sql
 	}
 	return &pb.GetWorkflowDataResponse{Version: version}, nil
 }
+
+func isApplicableToSend(context context.Context, wfContext *pb.WorkflowContext, workerID string, sdb *sql.DB) bool {
```

Comment:
It will be amazing to get in a written format when a worker can receive workload (I think that's what this function does). 

This function also requires a test case with a few cases (table testing please).

Reply:
Sure, I can write a comment over it. And regarding the test cases, we have not written unit test or Table test for any of the  function. We are planning to write all those tests at once by creating a new issue in which we will list out the functions and then we can track the progress over that.

Reply:
Are we in a rush with this? otherwise tested and documented atomic unit of works are awesome!

Reply:
Yes, I agree with you of testing the unit of work are necessary. But we are surely in a bit of rush for this one. Since this is required in master branch as soon as possible. I will create another issue which will also be related to the #196 . I will try to cover that in to the fix of that issue.

Reply:
I think this is one of those methods that definitely needs 100% test coverage